containers-contrib/src/Map/Contrib/Laws.hs

```diff
@@ -1,11 +1,71 @@
-{-# LANGUAGE MonoLocalBinds #-}
-{-# LANGUAGE TypeFamilies   #-}
+{-# LANGUAGE TypeFamilies #-}
 
 module Map.Contrib.Laws
-       ( singletonFromList
+       ( nullEmpty
+       , emptyZeroSized
+       , singletonOneSized
+       , memberEmptyFalse
+       , newMemberExists
+       , newMemberYieldsValidValue
+       , lookupDefaultEmpty
+       , listToSingleton
+       , singletonFromList
+       , keysOfSingleton
+       , elemsOfSingleton
+       , emptyInsertSingleton
+       , insertDeleteEmpty
+       , updateNothingRemoves
+       , updateJustId
+       , alterCanInsert
        ) where
 
-import Map (Key, fromList, singleton)
+import Map
+import Prelude hiding (lookup, null)
+
+nullEmpty :: Bool
+nullEmpty = null empty
```

Comment:
I think we can safely remove all duplicating properties with `containers-contrib-readonly`. Currently all non-readonly maps also implement `readonly` interface, so no need to duplicate properties.

Reply:
Did this and feeling great. This is my first time working with backpack, so did not know how to organize those different cabal files, but now I figured it out